pkg/collection/ducks_test.go

```diff
@@ -69,36 +70,75 @@ func makeCRDAnnotated(group, kind string, versions map[string]bool, labels, anno
 
 func TestNewDuckHunter(t *testing.T) {
 	tests := map[string]struct {
+		mapper   ResourceMapper
 		versions []v1alpha1.DuckVersion
 		want     *duckHunter
 	}{
-		"one versions": {
+		"no defaultVersions": {
+			want: &duckHunter{
+				mapper:          NewResourceMapper(nil),
+				defaultVersions: []string{},
+				ducks:           map[string][]v1alpha1.ResourceMeta{},
+			},
+		},
+		"one defaultVersions": {
 			versions: []v1alpha1.DuckVersion{
 				{Name: "v1"},
 			},
 			want: &duckHunter{
-				versions: []string{"v1"},
+				mapper:          NewResourceMapper(nil),
+				defaultVersions: []string{"v1"},
 				ducks: map[string][]v1alpha1.ResourceMeta{
 					"v1": {},
 				},
 			},
-		}, "three versions": {
+		},
+		"non nil mapper": {
+			versions: []v1alpha1.DuckVersion{
+				{Name: "v1"},
+			},
+			mapper: NewResourceMapper([]*metav1.APIResourceList{{
+				GroupVersion: "teach.me.how/v2",
+				APIResources: []metav1.APIResource{{
+					Kind:       "Ducky",
+					Name:       "duckies",
+					Namespaced: false,
+				}},
+			}}),
+			want: &duckHunter{
+				mapper: NewResourceMapper([]*metav1.APIResourceList{{
+					GroupVersion: "teach.me.how/v2",
+					APIResources: []metav1.APIResource{{
+						Kind:       "Ducky",
+						Name:       "duckies",
+						Namespaced: false,
+					}},
+				}}),
+				defaultVersions: []string{"v1"},
+				ducks: map[string][]v1alpha1.ResourceMeta{
+					"v1": {},
+				},
+			},
+		},
+		"three defaultVersions": {
 			versions: []v1alpha1.DuckVersion{{Name: "v1"}, {Name: "v2"}, {Name: "v3"}},
 			want: &duckHunter{
-				versions: []string{"v1", "v2", "v3"},
+				mapper:          NewResourceMapper(nil),
+				defaultVersions: []string{"v1", "v2", "v3"},
 				ducks: map[string][]v1alpha1.ResourceMeta{
 					"v1": {},
 					"v2": {},
 					"v3": {},
 				},
 			},
 		},
-		"overlapping versions": {
+		"overlapping defaultVersions": {
```

Comment:
How do v1,v2 overlap?

Reply:
overlap: `{Name: "v2"}, {Name: "v2"}`

Reply:
But those are "versions", not "defaultVersions" contrary to what the test name suggests.
Am I missing something?

Reply:
ah yeah I just need to update the test var name, I added `default` to versions this PR to make it more clear what the hunter has internally. Default versions are only settable in the constructor.

Reply:
ah shoot, it was merged. I fixed it for ya 😅

Reply:
Next PR 😄